tests/test_callbacks.py

```diff
@@ -0,0 +1,332 @@
+                'Rectangle',
+                'Roundrectangle',
+                'Barrel',
+                'Diamond',
+                'Pentagon',
+                'Star',
+                'Tag',
+                'Vee',
+                'Ellipse'
+            ],
+            prefix='NodeShape'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-node-color',
+            dir_name='style',
+            options=[
+                'pink',
+                'sky blue',
+                'rgb(186,44,162)',
+                '#def229'
+            ],
+            prefix='NodeColor'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-node-border-width',
+            dir_name='style',
+            options=['2'],
+            save=False
+        )
+
+        create_input_and_save(
+            css_selector='input#input-node-border-color',
+            dir_name='style',
+            options=[
+                'pink',
+                'sky blue',
+                'rgb(186,44,162)',
+                '#def229'
+            ],
+            prefix='BorderColor'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-node-border-width',
+            dir_name='style',
+            options=['5', '2'],
+            prefix='NodeBorderWidth'
+        )
+
+        create_input_and_save(
+            css_selector='input#dropdown-node-border-style',
+            dir_name='style',
+            options=[
+                'Dashed',
+                'Dotted',
+                'Double',
+                'Solid',
+            ],
+            prefix='NodeBorderStyle'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-node-padding',
+            dir_name='style',
+            options=['5px'],
+            prefix='NodePadding'
+        )
+
+        create_input_and_save(
+            css_selector='input#dropdown-node-padding-relative-to',
+            dir_name='style',
+            options=['Width', 'Height', 'Average', 'Min', 'Max'],
+            prefix='NodePaddingRelativeTo'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-edge-line-width',
+            dir_name='style',
+            options=['10', '1', '3'],
+            prefix='LineWidth'
+        )
+
+        create_input_and_save(
+            css_selector='input#dropdown-edge-curve-style',
+            dir_name='style',
+            options=['Haystack', 'Segments', 'Unbundled-bezier', 'Bezier'],
+            prefix='EdgeCurveStyle'
+        )
+
+        create_input_and_save(
+            css_selector='input#input-edge-line-color',
+            dir_name='style',
+            options=[
+                'pink',
+                'sky blue',
+                'rgb(186,44,162)',
+                '#def229'
+            ],
+            prefix='EdgeColor'
+        )
+
+        # Modify Edge Styles
+        click_button_and_save(
+            name_to_xpaths={
```

Comment:
huh, I've never seen xpaths before. I usually deal with CSS selectors (`#radio-edge-line-style label`) - is there a reason you used xpaths?

Reply:
I suppose the `:nth-child(3)` might not be supported?

Reply:
I believe that the CSS selector for buttons is ambiguous (there are multiple components that can be selected), so I went with xpath since it's more reliable in most situations.